plugins/src/main/java/opengrok/auth/plugin/util/FilterUtil.java

```diff
@@ -25,12 +25,59 @@
 
 import opengrok.auth.plugin.entity.User;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
 public class FilterUtil {
 
     private FilterUtil() {
         // utility class
     }
 
+    private static final String LOWER_CASE = "toLowerCase";
+    private static final String UPPER_CASE = "toUpperCase";
+
+    public static void checkTransforms(Map<String, String> transforms) {
+        Set<String> possibleTransforms = new HashSet<>(Arrays.asList(LOWER_CASE, UPPER_CASE));
+        for (String transform : transforms.values()) {
+            if (!possibleTransforms.contains(transform)) {
+                throw new UnsupportedOperationException(String.format("invalid transform: %s", transform));
+            }
+        }
+    }
+
+    static String doTransform(String value, String transform) {
+        switch (transform) {
+            case LOWER_CASE:
+                return value.toLowerCase(Locale.ROOT);
+            case UPPER_CASE:
+                return value.toUpperCase(Locale.ROOT);
+            default:
+                throw new UnsupportedOperationException(String.format("transform '%s' is unsupported", transform));
```

Comment:
did you try what happens when you try to load/use plugin with such settings?

Reply:
you mean testing the valid values or the exception case ?

Reply:
tested both `toUpperCase` and invalid transform. The plugin loaded with the former just fine, the latter failed with exception as expected:
```
14-Oct-2019 14:12:55.976 SEVERE [http-nio-8080-exec-564] org.opengrok.indexer.authorization.AuthorizationPlugin.load Plugin "opengrok.auth.plugin.LdapFilterPlugin" has failed while loading with exception:
        java.lang.UnsupportedOperationException: invalid transform: toFoo
                at opengrok.auth.plugin.util.FilterUtil.checkTransforms(FilterUtil.java:46)
                at opengrok.auth.plugin.LdapFilterPlugin.loadTransforms(LdapFilterPlugin.java:105)
                at opengrok.auth.plugin.LdapFilterPlugin.load(LdapFilterPlugin.java:91)
                at org.opengrok.indexer.authorization.AuthorizationPlugin.load(AuthorizationPlugin.java:121)
                at org.opengrok.indexer.authorization.AuthorizationStack.load(AuthorizationStack.java:148)
                at org.opengrok.indexer.authorization.AuthorizationStack.load(AuthorizationStack.java:148)
                at org.opengrok.indexer.authorization.AuthorizationFramework.loadAllPlugins(AuthorizationFramework.java:293)
                at org.opengrok.indexer.authorization.AuthorizationFramework.afterReload(AuthorizationFramework.java:357)
                at org.opengrok.indexer.framework.PluginFramework.reload(PluginFramework.java:379)
...
```